### PR TITLE
Expanded mgm user model

### LINKED ISSUE #8: Expand on User Model
Here's some other fields we could discuss and consider adding:
 - `PastDrifts` - array of drift object IDs that the User has participated in
 - `Friends` - array of user object IDs that the User is friends with
 - `Username` - string indicating User's unique username; it might be preferable for users to be identified by username rather than email
 - `FriendRequests` - array of user object IDs that indicate the User's pending friend requests
 - `CurrentDrift` - object including status on whether or not the User is currently in a drift, along with the drift's object ID

### PR DESCRIPTION
**Pull request by aidenszeto**
# Contributors
@aidenszeto 

# Reviewers
@brandonLi8 
@teekenzie 

# Relevant issue
Closes #8 

# Summary of change
- Updated user model to include the fields listed in #8

# Testing/Verification
- Server must compile
- Model fields must accurately represent fields listed in #8

### CODE REVIEW

**Comment by teekenzie** [comment]
Why do we need the 0 in the array. I don't think we need those right?

**Comment by aidenszeto** [comment]
> Why do we need the 0 in the array. I don't think we need those right?

I think the 0 defaults it to an empty array? I'm pretty sure array sizes have to be known by compiler at compile time too

**Comment by teekenzie** [comment]
> I think the 0 defaults it to an empty array? I'm pretty sure array sizes have to be known by compiler at compile time too
 
On second thought, you actually have to remove those. You are basically telling Go that we can only have a 0 length array placed into there. 

By not having any number, you could create a dynamic slice and it would still be treated as an array in mongo database

**Comment by aidenszeto** [comment]
> > I think the 0 defaults it to an empty array? I'm pretty sure array sizes have to be known by compiler at compile time too
> 
> On second thought, you actually have to remove those. You are basically telling Go that we can only have a 0 length array placed into there.
> 
> By not having any number, you could create a dynamic slice and it would still be treated as an array in mongo database

Ok yeah I forgot that you can have dynamic slices in Go, changes should be pushed

**Comment by teekenzie** [comment]
Also, since you have slice involved in your struct, please also include the creating function. For how to write this function, please refer to the demo-app

**Comment by aidenszeto** [comment]
> Also, since you have slice involved in your struct, please also include the creating function. For how to write this function, please refer to the demo-app

That's something we can do in a different ticket. This issue is for expanding the user model

**Comment by teekenzie** [comment]
> That's something we can do in a different ticket. This issue is for expanding the user model

I think this is something that should be included in this ticket. Since this is simply making sure that we won't be running into errors when inserting into database.

**Comment by aidenszeto** [comment]
> > That's something we can do in a different ticket. This issue is for expanding the user model
> 
> I think this is something that should be included in this ticket. Since this is simply making sure that we won't be running into errors when inserting into database.

**Comment by aidenszeto** [comment]
> > That's something we can do in a different ticket. This issue is for expanding the user model
> 
> I think this is something that should be included in this ticket. Since this is simply making sure that we won't be running into errors when inserting into database.

I think we should just merge this in and include the route in a different PR if we wanna stick to one ticket = one PR

File: models/user_model.go [hunk 1]
```diff
@@ -9,5 +12,10 @@ type User struct {
 	EmailAddress     string `bson:"email_address" json:"email_address" binding:"required"`
```
Comment (brandonLi8) [inline models/user_model.go]: Can we decide on using camel case for json? Seems like snake case is a bit out of place and [google recommends camel case](https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format).

**Review by brandonLi8 — CHANGES_REQUESTED** [review]
> I think we should just merge this in and include the route in a different PR if we wanna stick to one ticket = one PR

I actually agree with @teekenzie on this one. Adding the creating function is associated with your changes and this ticket.

**Comment by aidenszeto** [comment]
> > I think we should just merge this in and include the route in a different PR if we wanna stick to one ticket = one PR
> 
> I actually agree with @teekenzie on this one. Adding the creating function is associated with your changes and this ticket.

Ok yeah I misinterpreted what was meant by the creating function. Changes are pushed.

**Review by teekenzie — APPROVED** [review]

**Review by brandonLi8 — APPROVED** [review]
lgtm

**Pull request merged**